apps/examples/testcase/ta_tc/media/utc/utc_media_mediarecorder.cpp

```diff
@@ -107,62 +107,76 @@ static void utc_media_MediaRecorder_setVolume_p(void)
 {
 	uint8_t prev, volume;
 	MediaRecorder mr;
-	unique_ptr<FileOutputDataSource> dataSource = unique_ptr<FileOutputDataSource>(new FileOutputDataSource(channels, sampleRate, pcmFormat, filePath));
 	mr.create();
-	mr.setDataSource(std::move(dataSource));
-	mr.prepare();
 	mr.getVolume(&prev);
-
-	TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", mr.setVolume(10), RECORDER_OK, goto cleanup);
-	mr.getVolume(&volume);
-	TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", volume, 10, goto cleanup);
+	if (mr.setVolume(prev) == RECORDER_ERROR_DEVICE_NOT_SUPPORTED) {
+		printf("device doesn not support adjust volume, it using default\n");
+		TC_ASSERT_NEQ_CLEANUP("utc_media_mediarecorder_setVolume", mr.setVolume(prev + 1), RECORDER_OK, goto cleanup);		
+		TC_SUCCESS_RESULT();
+	} else {
+		TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", mr.setVolume(10), RECORDER_OK, goto cleanup);
+		mr.getVolume(&volume);
+		TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", volume, 10, goto cleanup);
+		TC_SUCCESS_RESULT();
+	}
 
 cleanup:
-	mr.setVolume(prev);
-	mr.unprepare();
 	mr.destroy();
-	TC_SUCCESS_RESULT();
 }
 
 static void utc_media_MediaRecorder_setVolume_n(void)
 {
-	uint8_t prev;
+	uint8_t prev, volume;
 	MediaRecorder mr;
-	unique_ptr<FileOutputDataSource> dataSource = unique_ptr<FileOutputDataSource>(new FileOutputDataSource(channels, sampleRate, pcmFormat, filePath));
-	TC_ASSERT_EQ("utc_media_mediarecorder_setVolume", mr.setVolume(10), RECORDER_ERROR_NOT_ALIVE);
 	mr.create();
-	mr.setDataSource(std::move(dataSource));
-	mr.prepare();
 	mr.getVolume(&prev);
-	TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", mr.setVolume(11), RECORDER_OK, goto cleanup);
-
+	if (mr.setVolume(prev) == RECORDER_ERROR_DEVICE_NOT_SUPPORTED) {
+		mr.setVolume(prev + 1);
+		mr.getVolume(&volume);
+		TC_ASSERT_NEQ_CLEANUP("utc_media_mediarecorder_setVolume", prev, volume, goto cleanup);		
+		TC_SUCCESS_RESULT();
+	} else {
+		TC_ASSERT_EQ_CLEANUP("utc_media_mediarecorder_setVolume", mr.setVolume(11), RECORDER_OK, goto cleanup);
+	}
+	
+	TC_SUCCESS_RESULT();
 cleanup:
-	mr.setVolume(prev);
-	mr.unprepare();
 	mr.destroy();
-	TC_SUCCESS_RESULT();
 }
 
 static void utc_media_MediaRecorder_getVolume_p(void)
 {
 	uint8_t volume;
+	uint8_t maxVolume;
```

Comment:
Where is this maxVolume used..?

Reply:
Removed, I had a plan that provide new api to get max Volume, I will request to you =D